### PR TITLE
Tests: Updates to QUnit version 1.17.1

### PR DESCRIPTION
**Pull request by apsdehal**
Upgrades QUnit version to 1.17.1

### CODE REVIEW

**Comment by leobalter** [comment]
Maybe you wanna replace [this line](https://github.com/jquery/jquery-mobile/blob/master/tests/unit/init/init_core.js#L4) with a `QUnit.config.autostart = false;`

Ref: http://api.qunitjs.com/QUnit.config/

**Comment by apsdehal** [comment]
Thanks @leobalter. Fixed.

**Comment by leobalter** [comment]
Checking the CI issue.

`QUnit.start()` throws an error if called more than once to start a test (when using the `QUnit.config.autostart = false;` setting).

Then I searched and found two QUnit.start() being called:

- One from the [`$.testHelper.asyncLoad`](https://github.com/jquery/jquery-mobile/blob/master/tests/jquery.testHelper.js#L22) called at the [`init/index.html`](https://github.com/jquery/jquery-mobile/blob/master/tests/unit/init/index.html#L18-L20)
- The second one is called at [`init/init_core.js`](https://github.com/jquery/jquery-mobile/blob/master/tests/unit/init/init_core.js#L25)

You need to remove this second call, as the helper is already doing it's job. Regarding that, it's more accurate to place `QUnit.config.autostart = false;` before the $.testHelper.asyncLoad call.

**Comment by apsdehal** [comment]
Oh, I didn't notice that. Will push fix soon.

**Comment by gabrielschulhof** [comment]
@arschmitz I've added the priority flag to this PR because it's a prerequisite for using your updated version of qunit-assert-classes - which we want to use for all classes PRs, so they all depend on this PR.

File: tests/unit/field-contain/fieldContain_events.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 		$( "#ignore-container-tests" ).trigger( "create" );
 
-		deepEqual( $ignored.attr( "class" ), undefined, "ignored div does not have field contain class" );
+		ok( !$ignored.hasClass( "ui-field-contain" ), "ignored div does not have field contain class" );
```
Comment (gabrielschulhof) [inline tests/unit/field-contain/fieldContain_events.js]: Long line (needs to be below 100 characters (one tab counts as 4 characters))

Comment (apsdehal) [inline tests/unit/field-contain/fieldContain_events.js]: Is this convention mostly followed? I see lines with 150 characters. :-)

**Comment by arschmitz** [comment]
:+1: @gabrielschulhof feel free to rebase and land this.

**Pull request closed without merging**